pkg/operator/ceph/version/version.go

```diff
@@ -65,6 +67,10 @@ var (
 	// So we need to detect the build version change
 	buildVersionPattern = regexp.MustCompile(`ceph version (\d+)\.(\d+)\.(\d+)\-(\d+)`)
 
+	buildSuffixPattern = regexp.MustCompile(`ceph version (\d+)\.(\d+)\.(\d+)\-(\d+).(\S+)`)
+
+	imagePattern = regexp.MustCompile(`\(([^)]+)\)`)
```

Comment:
Does this match everything from the first character to the first whitespace? Or what will match? Can you add a comment about it? I like this idea of getting the entire version string and not worry about the fields. Then I don't think we need to get the buildSuffix, we can just use this full image name and if the ceph build version comes up with even more unanticipated fields we will already detect it instead of needing more fields.

Reply:
This would only return the commit ID. 
For example, it will return `b8e1f91c99491fb2e5ede748a1c0738ed158d0f5` from the source `ceph version 14.2.11-139.el8cp (b8e1f91c99491fb2e5ede748a1c0738ed158d0f5) nautilus (stable)`.  I was trying to address the [comment](https://github.com/rook/rook/pull/8060#pullrequestreview-677678759) with this comparison.

Reply:
Ah that makes sense. That image hash will be a good representation of whether anything changed. So if we get this image and it changed, we would trigger the child controllers, it should fully address the issue. There is no need to extract the buildSuffix now, right?

Reply:
Yes, checking commit hash should suffice and we can ignore the `buildSuffix`. This should fix the case when build number didn't change due to a hotfix.